Examples/Google/SpeechToText/Sources/SpeechService.swift

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPC
+
+// Track whether we are currently streaming or not
+enum State {
+  case idle
+  case streaming(BidirectionalStreamingCall<Google_Cloud_Speech_V1_StreamingRecognizeRequest, Google_Cloud_Speech_V1_StreamingRecognizeResponse>)
+}
+
+final class SpeechService {
+  
+  // Generated SpeechClient for making calls
+  private var client: Google_Cloud_Speech_V1_SpeechClient
+
+  private var call: BidirectionalStreamingCall<Google_Cloud_Speech_V1_StreamingRecognizeRequest, Google_Cloud_Speech_V1_StreamingRecognizeResponse>?
+  
+  // Track if we are streaming or not
+  private var state: State {
+    return call != nil ? .streaming(call!) : .idle
+  }
+  
+  init() {
+    guard !Constants.apiKey.isEmpty else {
+      fatalError("Please refer to the README on how to configure your API Key properly.")
+    }
+    
+    // Make EventLoopGroup for the specific platform (NIOTSEventLoopGroup for iOS)
+    let group = PlatformSupport.makeEventLoopGroup(loopCount: 1)
+    
+    // Create connection channel with our group, host, and port
+    let channel = ClientConnection
+      .secure(group: group)
+      .connect(host: "speech.googleapis.com", port: 443)
+
+    // Specify call options to be used for gRPC calls
+    let callOptions = CallOptions(customMetadata: [
+      "x-goog-api-key": Constants.apiKey
+    ])
+
+    // Now we have a client!
+    client = Google_Cloud_Speech_V1_SpeechClient(channel: channel, defaultCallOptions: callOptions)
+  }
+  
+  func stream(_ data: Data,
+              completion: ((Google_Cloud_Speech_V1_StreamingRecognizeResponse) -> Void)? = nil) {
+
+    switch state {
+    case .idle:
+      // Initialize the bidirectional stream
+      call = client.streamingRecognize { (response) in
+        // Message received from Server, execute provided closure from caller
+        completion?(response)
+      }
```

Comment:
We need to update `self.state` after this as well: `self.state = .streaming(call)`